e2e/helpers/http.ts

```diff
@@ -1,141 +1,107 @@
-import { createHmac } from 'crypto'
-import { fetchWithExponentialTimeout } from './srcFetch'
+import { Page, expect } from '@playwright/test'
+import { navigateToHomepageIfPageHasNotBeenLoaded } from './localStorage'
 
-export const BASE64_ENCODING = 'base64'
-export const DEFAULT_GETDELETE_HEADERS = {
-  Accept: '*/*',
-  'User-Agent': 'SynapseWebClient',
-}
-export const DEFAULT_POST_HEADERS = {
-  ...DEFAULT_GETDELETE_HEADERS,
-  Accept: 'application/json; charset=UTF-8',
-  'Content-Type': 'application/json; charset=UTF-8',
-}
-
-function generateDigitalSignature(
-  data: string,
-  base64EncodedSecretKey: string,
-) {
-  const hash = createHmac(
-    'sha1',
-    Buffer.from(base64EncodedSecretKey, BASE64_ENCODING),
-  )
-    .update(data)
-    .digest(BASE64_ENCODING)
-  return hash
-}
-
-function getDigitalSignature(
-  uri: string,
-  username: string,
-  base64EncodedSecretKey: string,
-) {
-  const timestamp = new Date().toISOString()
-  const signature = generateDigitalSignature(
-    `${username}${uri}${timestamp}`,
-    base64EncodedSecretKey,
-  )
-  return {
-    userId: username,
-    signatureTimestamp: timestamp,
-    signature: signature,
-  }
-}
-
-export function getUserIdFromJwt(token: string) {
-  const payload = JSON.parse(
-    Buffer.from(token.split('.')[1], BASE64_ENCODING).toString(),
-  )
-  return payload.sub
+export enum BackendDestinationEnum {
+  REPO_ENDPOINT,
+  PORTAL_ENDPOINT,
```

Comment:
Replicate [`BackendDestinationEnum`](https://github.com/Sage-Bionetworks/synapse-web-monorepo/blob/bc092bdea2b0fd67f6bb897fd6f22f556e3a0ab2/packages/synapse-react-client/src/utils/functions/getEndpoint.ts#L1), so that the endpoint can be passed to `doGet`, etc... without evaluating the page to access SRC.